### DISCUSSION TITLE
Should I have separate .fvmrc files for all packages within the same project?

### DISCUSSION THREAD

**Discussion by iamongit** [leoafarias/fvm#659] [Q&A] [answered]
Before updating fvm while I was still using fvm_config.json, in my GIT workflows I could just do fvm install in the root folder and all the packages in the project would use the same flutter version. After updating fvm recently, doing fvm install in root folder installs the flutter using the version specified in .fvmrc file but once it changes directory to go to a different package within the same project I get "flutter not found".

I can fix this by:
 - Setting fvm global version after fvm install in root folder.
 - Having .fvmrc files for all packages with same flutter version as specified in .fvmrc of root folder.

Should I be setting the global version or have separate .fvmrc files which means maintaining that version and synchronizing it with the version of .fvmrc in root folder?

I only have this issue on GIT as locally I've flutter version added to path.

**Comment by leoafarias** [comment]
@iamongit, you don't need to have a .fvmrc for each package. But you need to have a pubspec at the root of the directory that you want the .fvmrc in

**Comment by iamongit** [comment]
> @iamongit, you don't need to have a .fvmrc for each package. But you need to have a pubspec at the root of the directory that you want the .fvmrc in

Thanks, I am still a bit unclear on this. If I have a project containing e.g. 3 flutter packages each with their own pubspec file, earlier I only needed one fvm_config.json at the root folder which contained these3 package folders. Now do I need 3 separate .fvmrc files for the 3 packages?
Here's the folder structure structure:
``` 
- Top Level Package Folder Containing 3 flutter packages
-- Folder 1 containing package 1 with its pubspec file
-- Folder 2 containing Package 2 with its pubspec file
-- Folder 3 containing Package 3 with its pubspec file 
```
What would be the optimal way of maintaining this instead of having 3 .fvmrc files and having to maintain the version in all 3 files?
Right now for my git workflow I have only 1 .fvmrc in top level folder and I do fvm global after fvm install to have fvm flutter available in individual projects on directory change.

**Reply by leoafarias** [reply, marked as answer]
You need to create a pubspec.yaml at the root, you can do something like

pubspec.yaml
```yaml
name: workspace_name
```

and now you can have a .fvmrc on that directory

**Reply by iamongit** [reply]
Thank you, appreciate it. I added a pubspec.yaml to the root folder (which also has the only .fvmrc file) with just the name, description and environment and now I don't need to set a global version. I suppose it doesn't even need the environment or description in there.

Somehow I thought pubspec.yaml file only applied to an actual flutter project folder including all the dependencies and everything.

**Reply by leoafarias** [reply]
@iamongit  you are not wrong about the pubspec.yaml. You have to add the name due to the parsing. Normally, a monorepo package like 'melos,' for example, does require a workspace pubspec for shared dependencies and other things.
